### PR TITLE
[protobuf-c] Fix build error for windows

### LINKED ISSUE #29677: [protobuf-c] Build error
Package: protobuf-c[core,tools]:x86-windows -> 1.4.0#1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.29.30148.0
-    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: ce8d96a7c 2023-02-14 (22 hours ago)

**To Reproduce**

`vcpkg install protobuf-c[tools]`

**Failure logs**

```
-- Using cached protobuf-c-protobuf-c-v1.4.0.tar.gz.
-- Cleaning sources at D:/vcpkg/buildtrees/protobuf-c/src/v1.4.0-de9c7632f7.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/protobuf-c-protobuf-c-v1.4.0.tar.gz
-- Applying patch fix-crt-linkage.patch
-- Using source at D:/vcpkg/buildtrees/protobuf-c/src/v1.4.0-de9c7632f7.clean
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Fixing pkgconfig file: D:/vcpkg/packages/protobuf-c_x86-windows/lib/pkgconfig/libprotobuf-c.pc
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Fixing pkgconfig file: D:/vcpkg/packages/protobuf-c_x86-windows/debug/lib/pkgconfig/libprotobuf-c.pc
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "protoc-c":

      "D:/vcpkg/packages/protobuf-c_x86-windows/bin/protoc-c.exe" does not exist
Call Stack (most recent call first):
  ports/protobuf-c/portfile.cmake:28 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)



```

**Additional context**

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/29677
```
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "protoc-c":

      "D:/vcpkg/packages/protobuf-c_x86-windows/bin/protoc-c.exe" does not exist
Call Stack (most recent call first):
  ports/protobuf-c/portfile.cmake:28 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)
```
```
upstream code:
makelist.md
install-exec-hook:
	rm -f $(DESTDIR)$(bindir)/protoc-c
	ln -s protoc-gen-c $(DESTDIR)$(bindir)/protoc-c
```

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Tested usage successfully by the protobuf-c:x64-windows triplet.

### CODE REVIEW

**Comment by gvanem** [comment]
I need this `protoc-c.exe` prorgram so I tried this patch, but it does not produce this program. So what's the point of this PR?

**Comment by dg0yt** [comment]
> I need this `protoc-c.exe` prorgram so I tried this patch, but it does not produce this program. So what's the point of this PR?

I guess the program you are looking for is called `protoc-gen-c` now.
https://github.com/protobuf-c/protobuf-c/blob/abc67a11c6db271bedbb9f58be85d6f4e2ea8389/ChangeLog#L139-L143

**Comment by gvanem** [comment]
**dg0yt**> I guess the program you are looking for is called protoc-gen-c now.

I tried `proto-gen-c.exe`, but it read from `stdin`!? Not the same as `protoc-c.exe` AFAICS.

**quote**>  but it's a symlink to the protoc-gen-c binary.

Symlink?? I'm on effin Windows!

**Comment by jimwang118** [comment]
> I need this `protoc-c.exe` prorgram so I tried this patch, but it does not produce this program. So what's the point of this PR?

In the upstream MakeFile.md file it looks like this:
```
install-exec-hook:
	rm -f $(DESTDIR)$(bindir)/protoc-c
	ln -s protoc-gen-c $(DESTDIR)$(bindir)/protoc-c
```
This means `protoc-c.exe` is a soft link to `protoc-gen-c.exe`.

**Comment by MonicaLiu0311** [comment]
And I found that `protoc-c` was added to `tools` in [this PR](https://github.com/microsoft/vcpkg/pull/28403/files#diff-5e9e01f4eb04eee2ed520cde847acd38cb20df9722956fb2a30539c68aec10b4R29), @autoantwort , could you please help us to see if this change is reasonable? or any advice? Thanks in advance🌻

**Comment by autoantwort** [comment]
@MonicaLiu0311 Sorry I don't have any experience with this port and don't know if the change is reasonable.

**Comment by aviborg** [comment]
> **dg0yt**> I guess the program you are looking for is called protoc-gen-c now.
> 
> I tried `proto-gen-c.exe`, but it read from `stdin`!? Not the same as `protoc-c.exe` AFAICS.
> 
> **quote**> but it's a symlink to the protoc-gen-c binary.
> 
> Symlink?? I'm on effin Windows!

@dg0yt  `protoc-c.exe` is not building on Windows at all with vcpkg, that's why this issue was created. Or can you tell us how you manage to do `vcpkg.exe install protobuf-c[tools]` on Windows without getting an error? 

Note that this Protobuf-c `protoc-gen-c.exe` is a plugin to Google Protobuf `protoc.exe`. 
This is not to be confused with the old standalone Protobuf-c `protoc-c.exe`. Some archived info here: https://code.google.com/archive/p/protobuf-c/wikis/Examples.wiki

The Google Protobuf `protoc.exe` generates .pb-c.c and .pb-c.h output files from your .proto input file. The --c_out options instructs `protoc.exe` to use the Protobuf-c (`protobuf-gen-c.exe`) plugin.

```
protoc.exe --c_out=. example.proto
```
The above will search the PATH for the `protoc-gen-c.exe` and generate C protocol buffers.

I think @autoantwort commited the addition of the `TOOL protoc-c` by mistake.

**Comment by autoantwort** [comment]
@aviborg My PR fixed the osx build of protobuf-c, but it seems it broke the windows build 
image

**Comment by autoantwort** [comment]
> Is the protoc-c executable really needed?

No, it should be ok to simply delete it

**Comment by aviborg** [comment]
> > Is the protoc-c executable really needed?
> 
> No, it should be ok to simply delete it

@MonicaLiu0311 I think you could approve this PR now.

**Comment by autoantwort** [comment]
No it leaves a broken symlink behind
image

**Comment by jimwang118** [comment]
> No it leaves a broken symlink behind image

The soft link left by `protoc-c` has been removed.
image

File: ports/protobuf-c/portfile.cmake [hunk 1]
```diff
@@ -37,6 +37,8 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 # Remove duplicate PDB files (vcpkg_copy_pdbs already copied them to "bin")
 file(REMOVE "${CURRENT_PACKAGES_DIR}/lib/protobuf-c.pdb")
 file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/lib/protobuf-c.pdb")
-
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
Comment (dg0yt) [inline ports/protobuf-c/portfile.cmake:40]: IMO the required condition is `if(NOT VCPKG_TARGET_IS_WINDOWS)`.
- On Windows, `protoc-c` isn't generated, and cleanup is handled by `vcpkg_copy_tools`.
- On non-windows, the directory is never needed in vcpkg. It must be deleted at this point regardless of linkage, due to `protoc-c` symlink.

**Pull request merged**